src/tasks/Microsoft.NET.Sdk.WebAssembly.Pack.Tasks/BootJsonBuilderHelper.cs

```diff
@@ -68,6 +68,8 @@ static void AddDictionary(StringBuilder sb, Dictionary<string, string>? res)
                 return bootConfig.resources.wasmSymbols ??= new();
             else if (resourceName.StartsWith("icudt", StringComparison.OrdinalIgnoreCase))
                 return bootConfig.resources.icu ??= new();
+            else if (resourceName.StartsWith("segmentation-rules", StringComparison.OrdinalIgnoreCase) && string.Equals(resourceExtension, ".json", StringComparison.OrdinalIgnoreCase))
```

Comment:
I wonder if there are any remaining places which need to be edited in SDK repo. @maraf

Reply:
IMO not in the build side. There are tests in SDK and aspnetcore repo, but as mentioned they most probably don't use hybrid globalization (anyway, it would be good to watch flow)

Reply:
`StartsWith` -> `Equals`.
Also, add the new file to https://github.com/dotnet/runtime/blob/e458d68a3879d968faa1da9de38876e23935c166/src/mono/wasm/Wasm.Build.Tests/ProjectProviderBase.cs#L362-L364

Reply:
Why not compare the whole filename (`resourceName`) with `"segmentation-rules.json"`?

Reply:
I was originally following the code style of above if conditions e.g., for `dotnet.native.wasm` https://github.com/dotnet/runtime/blob/22ba7d607bb1d9caa0db9afcdc47eb5cef641fcb/src/tasks/Microsoft.NET.Sdk.WebAssembly.Pack.Tasks/BootJsonBuilderHelper.cs#L63. However, I don't see why we couldn't compare whole filename for `segmentation-rules.json` since we don't expect any more substrings between the `segmentaiton-rules` name  and `json` extension.

Reply:
Exactly, the others are expecting fingerprint in filename